kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java

```diff
@@ -108,7 +108,11 @@ public KafkaProxy startup() throws InterruptedException {
         var bindFutures = networkBindings.stream()
                 .map(networkBinder -> networkBinder.bind(networkBinder.tls() ? tlsServerBootstrap : plainServerBootstrap))
                 .collect(Collectors.toSet());
-        bindFutures.forEach(channelFuture -> channelFuture.addListener(f -> LOGGER.info("Listening : {}", channelFuture.channel().localAddress())));
+        bindFutures.forEach(channelFuture -> channelFuture.addListener(f -> {
+            var channel = ((ChannelFuture) f).channel();
+            acceptorChannels.add(channel);
```

Comment:
I think this works for now but something about it bothers me (the shape of how we do blocking not the fix itself).

Reply:
Yeap, will have a go after merging this. It feels like we should be able to do something like collect up all the channels after the `sync` on all the bind futures and make a composed `allOf` type future out of all the closeFutures that we can block on without looping.